dhall/src/Dhall/Import.hs

```diff
@@ -506,20 +504,22 @@ loadImportWithSemanticCache
             let actualHash = Dhall.Crypto.sha256Hash bytesStrict
 
             if semanticHash == actualHash
-                then return ()
-                else do
-                    Status { _stack } <- State.get
-                    throwMissingImport (Imported _stack (HashMismatch {expectedHash = semanticHash, ..}))
+                then do
+                    let bytesLazy = Data.ByteString.Lazy.fromStrict bytesStrict
 
-            let bytesLazy = Data.ByteString.Lazy.fromStrict bytesStrict
+                    importSemantics <- case Dhall.Binary.decodeExpression bytesLazy of
+                        Left  err -> throwMissingImport (Imported _stack err)
+                        Right e   -> return e
 
-            importSemantics <- case Dhall.Binary.decodeExpression bytesLazy of
-                Left  err -> throwMissingImport (Imported _stack err)
-                Right e   -> return e
+                    return (ImportSemantics {..})
+                else do
+                    printWarning $ makeHashMismatchMessage semanticHash actualHash
```

Comment:
Perhaps extend the warning message to say that the interpreter has automatically fixed the cached import

Reply:
I updated PR description with new message